apps/pkeyutl.c

```diff
@@ -328,8 +350,8 @@ int pkeyutl_main(int argc, char **argv)
     }
 
     if (in != NULL) {
-        /* Read the input data */
-        buf_inlen = bio_to_mem(&buf_in, keysize * 10, in);
+        /* Read the input data. Raw input data has no maxlen limit */
+        buf_inlen = bio_to_mem(&buf_in, rawin ? -1 : keysize * 10, in);
```

Comment:
Probably if people are going to use this option they are going to expect to be able to provide arbitrarily long input files. Reading several Gb into memory might not be advisable! Probably we should stream this data in that case (i.e. use EVP_DigestSignUpdate/EVP_DigestVerifyUpdate).

Reply:
Good point - will move to stream mode...